packages/data-broker/src/tests/services/weather/WeatherService.test.js

```diff
@@ -0,0 +1,234 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { WeatherService } from '../../../services/weather';
+import {
+  createMockModelsStubWithMockEntity,
+  createWeatherApiStub,
+  createWeatherApiStubWithMockResponse,
+  getMockDataSourcesArg,
+  getMockOptionsArg,
+  getMockTypeArg,
+} from './WeatherService.stubs';
+import { mockNow, resetMocks } from './testutil';
+
+describe('WeatherService', () => {
+  afterEach(() => {
+    resetMocks();
+  });
+
+  describe('basic operation', () => {
+    it('returns analytics data when requesting data for dataElements', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApiResponse = {
+        data: [
+          {
+            precip: 23.6,
+            max_temp: 29.8,
+            min_temp: 24,
+            datetime: '2020-08-20',
+          },
+          {
+            precip: 5,
+            max_temp: 6,
+            min_temp: 7,
+            datetime: '2020-08-21',
+          },
+        ],
+        sources: [],
+      };
+
+      const mockApi = createWeatherApiStub(mockApiResponse);
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const actual = await service.pull(
+        [
+          {
+            model: {},
+            id: '12345_PRECIP',
+            code: 'PRECIP',
+            type: 'dataElement',
+            service_type: 'weather',
+            config: {},
+          },
+        ],
+        'dataElement',
+        getMockOptionsArg(),
+      );
+
+      expect(actual.results).to.deep.equal([
+        { dataElement: 'PRECIP', value: 23.6, organisationUnit: 'MELB', period: '20200820' },
+        { dataElement: 'PRECIP', value: 5, organisationUnit: 'MELB', period: '20200821' },
+      ]);
+    });
+
+    it('returns events data when requesting data for a dataGroup', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApiResponse = {
+        data: [
+          {
+            precip: 23.6,
+            max_temp: 29.8,
+            min_temp: 24,
+            datetime: '2020-08-20',
+          },
+          {
+            precip: 5,
+            max_temp: 6,
+            min_temp: 7,
+            datetime: '2020-08-21',
+          },
+        ],
+        sources: [],
+      };
+
+      const mockApi = createWeatherApiStub(mockApiResponse);
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const actual = await service.pull(
+        [
+          {
+            model: {},
+            id: 'SOME_DATA_GROUP_ID', // data group doesn't matter, it will return the data elements it knows about
```

Comment:
Which is maybe not a greeaat thing...

Reply:
Yeah? I don't really understand data groups fully yet, what do you recommend instead?

Reply:
I'd go with retrieving the data elements attached to the data group via the join table. May be a precedent for this in one of the other services

Reply:
Updated